R/generate_simulated_data.R

```diff
@@ -328,20 +328,59 @@ generate_simulated_data <- function(r_in_weeks = # nolint
     uot + ot + ht
   )[(uot + 1):(uot + ot + ht)]
 
+  # Also compute per capita hosps for each subpopulation
+  model_hosp_subpop_over_n <- matrix(
+    nrow = n_subpops,
+    ncol = (ot + ht)
+  )
+  for (i in 1:n_subpops) {
+    model_hosp_subpop_over_n[i, ] <- model$functions$convolve_dot_product(
+      p_hosp_days * new_i_over_n_site[i, ],
+      rev(inf_to_hosp),
+      uot + ot + ht
+    )[(uot + 1):(uot + ot + ht)]
+  }
+
 
   ## Add weekday effect on hospital admissions-------------------------------
   pred_hosp <- pop_size * model$functions$day_of_week_effect(
     model_hosp_over_n,
     day_of_week_vector,
     hosp_wday_effect
   )
+
+  pred_hosp_subpop <- matrix(
+    nrow = n_subpops,
+    ncol = (ot + ht)
+  )
+  for (i in 1:n_subpops) {
+    pred_hosp_subpop[i, ] <- pop_fraction[i] * pop_size *
+      model$functions$day_of_week_effect(
+        model_hosp_subpop_over_n[i, ],
+        day_of_week_vector,
+        hosp_wday_effect
+      )
+  }
+
+
   ## Add observation error---------------------------------------------------
   # This is negative binomial but could swap out for a different obs error
   pred_obs_hosp <- rnbinom(
     n = length(pred_hosp), mu = pred_hosp,
     size = 1 / ((params$inv_sqrt_phi_prior_mean)^2)
   )
 
+  pred_obs_hosp_subpop <- matrix(
+    nrow = n_subpops,
+    ncol = (ot + ht)
+  )
+  for (i in 1:n_subpops) {
+    pred_obs_hosp_subpop[i, ] <- rnbinom(
+      n = length(pred_hosp_subpop[i, ]), mu = pred_hosp_subpop[i, ],
+      size = 1 / ((params$inv_sqrt_phi_prior_mean)^2)
```

Comment:
A minor but important detail here is that the $\phi_i$ for the subpops will not in general be the same as the $\phi$ for the population as a whole: https://math.stackexchange.com/questions/2189105/sum-of-negative-binomial-distributed-r-v

I think if we provide options for users to fit subpopulation level observations with negative binomials, they should be able to infer subpopulation-specific $\phi$ values (perhaps hierarchically, perhaps with a population effect, etc).

For now, I would just let the user configure a `phi_subpop` vector for simulation.

Reply:
That makes a lot of sense and was thinking about that as I implemented! Ok so for now, we will specify phi_subpop. What this will mean is that there won't be a set phi, but that should be okay?